controllers/useraccount/useraccount_controller_test.go

```diff
@@ -477,6 +478,159 @@ func TestReconcile(t *testing.T) {
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+				},
+			},
+		}
+		noiseRb := &v1.RoleBinding{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "RoleBinding",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "noise-rb",
+				Namespace: UserSettingNS,
+			},
+		}
+		r, req, cl, _ := prepareReconcile(t, username, cfg, userAcc, preexistingUser, preexistingIdentity, configMap, role, rb, noiseCm, noiseRole, noiseRb)
+
+		t.Run("first reconcile deletes identity", func(t *testing.T) {
+			// given
+			// Set the deletionTimestamp
+			now := metav1.NewTime(time.Now())
+			userAcc.DeletionTimestamp = &now
+			err = r.Client.Update(context.TODO(), userAcc)
+			require.NoError(t, err)
+
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			assert.Equal(t, reconcile.Result{}, res)
+			require.NoError(t, err)
+
+			useraccount.AssertThatUserAccount(t, req.Name, cl).
+				HasConditions(notReady("Terminating", "deleting user/identity"))
+
+			// Check that the associated identity has been deleted
+			// when reconciling the useraccount with a deletion timestamp
+			assertIdentityNotFound(t, r, userAcc, config.Auth().Idp())
+			useraccount.AssertThatUserAccount(t, userAcc.Name, cl).HasConditions(terminating())
+
+			t.Run("second reconcile deletes user and its resources - configmap, role and rolebinding", func(t *testing.T) {
+				// when
+				res, err = r.Reconcile(context.TODO(), req)
+
+				// then
+				assert.Equal(t, reconcile.Result{}, res)
+				require.NoError(t, err)
+
+				useraccount.AssertThatUserAccount(t, req.Name, cl).
+					HasConditions(notReady("Terminating", "deleting user/identity"))
+
+				// Check that the associated configmap has been deleted
+				retrievedConfigMap := &corev1.ConfigMap{}
+				err := r.Client.Get(context.TODO(), types.NamespacedName{Name: resourceName, Namespace: UserSettingNS}, retrievedConfigMap)
+				require.Error(t, err)
+				assert.True(t, apierros.IsNotFound(err))
+				// Check that the associated role has been deleted
+				retrievedRole := &v1.Role{}
+				err = r.Client.Get(context.TODO(), types.NamespacedName{Name: resourceName, Namespace: UserSettingNS}, retrievedRole)
+				require.Error(t, err)
+				assert.True(t, apierros.IsNotFound(err))
+				// Check that the associated rolebinding has been deleted
+				retrievedRb := &v1.RoleBinding{}
+				err = r.Client.Get(context.TODO(), types.NamespacedName{Name: resourceName, Namespace: UserSettingNS}, retrievedRb)
+				require.Error(t, err)
+				assert.True(t, apierros.IsNotFound(err))
+
+				// Check that the noise resources are not deleted
+				retrievedNoiseCm := &corev1.ConfigMap{}
+				err = r.Client.Get(context.TODO(), types.NamespacedName{Name: "noise-object", Namespace: UserSettingNS}, retrievedNoiseCm)
+				require.NoError(t, err)
+				assert.Equal(t, noiseCm, retrievedNoiseCm)
```

Comment:
in a similar way, this can be replaced by:
```go
retrievedNoiseCm := &corev1.ConfigMap{}
AssertObject(t, cl, UserSettingNS, "noise-object", retrievedNoiseCm, func() {
	assert.Equal(t, noiseCm, retrievedNoiseCm)
})
```

Reply:
done in [d2b6226](https://github.com/codeready-toolchain/member-operator/pull/568/commits/d2b6226ae7d725105c82382ad27f43d0d7327e02)